pkg/testing/builder/clusterpodplacementconfig.go

```diff
@@ -34,3 +35,25 @@ func (p *ClusterPodPlacementConfigBuilder) WithLogVerbosity(logVerbosity common.
 func (p *ClusterPodPlacementConfigBuilder) Build() *v1beta1.ClusterPodPlacementConfig {
 	return p.ClusterPodPlacementConfig
 }
+
+func (p *ClusterPodPlacementConfigBuilder) WithNodeAffinityScoring(enabled bool) *ClusterPodPlacementConfigBuilder {
+	if p.Spec.Plugins == (plugins.Plugins{}) {
+		p.Spec.Plugins = plugins.Plugins{}
+	}
+	if p.Spec.Plugins.NodeAffinityScoring == nil {
+		p.Spec.Plugins.NodeAffinityScoring = &plugins.NodeAffinityScoring{}
+	}
+	p.Spec.Plugins.NodeAffinityScoring.BasePlugin.Enabled = enabled
+	return p
+}
+
+func (p *ClusterPodPlacementConfigBuilder) WithNodeAffinityScoringTerm(architecture string, weight int32) *ClusterPodPlacementConfigBuilder {
+	if p.Spec.Plugins.NodeAffinityScoring.Platforms == nil {
+		p.Spec.Plugins.NodeAffinityScoring.Platforms = []plugins.NodeAffinityScoringPlatformTerm{}
+	}
```

Comment:
```suggestion
```suggestion
	if p.Spec.Plugins.NodeAffinityScoring == nil {
	  p.Spec.Plugins.NodeAffinityScoring = &plugins.NodeAffinityScoring{}
	}
```

No need to initialize the slice, but to initialize the NodeAffinityScoring pointer.
```